dashboard/app/models/activity_section.rb

```diff
@@ -86,6 +87,23 @@ def update_script_levels(script_levels_data)
     end
   end
 
+  # Used for seeding from JSON. Returns the full set of information needed to uniquely identify this object.
+  # If the attributes of this object alone aren't sufficient, and associated objects are needed, then data from
+  # the seeding_keys of those objects should be included as well.
+  # Ideally should correspond to a unique index for this model's table.
+  # See comments on ScriptSeed.seed_from_json for more context.
+  #
+  # @param [ScriptSeed::SeedContext] seed_context - contains preloaded data to use when looking up associated objects
+  # @return [Hash<String, String>] all information needed to uniquely identify this object across environments.
+  def seeding_key(seed_context)
+    my_lesson_activity = seed_context.lesson_activities.select {|la| la.id == lesson_activity_id}.first
+    raise "No LessonActivity found for #{self.class}: #{my_key}, LessonActivity ID: #{lesson_activity_id}" unless my_lesson_activity
+    {
+      'activity_section.key': key,
+      'lesson_activity.key': my_lesson_activity.key
```

Comment:
Question for my own benefit: why do we need to include `lesson_activity.key` here? Shouldn't `activity_section.key` be enough to uniquely identify an activity section?

Reply:
Great question, which gets at the core of the new seeding strategy. The seeding_key must have enough information to uniquely identify the object within the script, but it also needs to tell us where to put the object (this part may be missing from the comments). this object is NOT nested within its parent (lesson_activity) in the serialized format, so we still need the `lesson_activity_key` in order to know where to put it.

With `m` models and `n` objects/instances of those models, this allows us to do `O(m)` DB queries to import all the data into each script, no matter how many objects there are. With a nested structure, the number of queries would increase with `n`.